File: Week1/Day5/Programs/nthFibonacci.java
public class nthFibonacci{
	public static int fib(int n){
		if(n==1){
			return 0;
		}
		if(n==2){
			return 1;
		}
		if(n==3){
			return 1;
		}
		int i=1;
		int j=1;
		int k=0;
		for(int r=2;r<n;r++){
			k=i+j;
			i=j;
			j=k;
		}
		return k;
	}
	public static void main(String[] args) {
		System.out.println(fib(4));
	}
}

File: Week1/Day5/Programs/nthFibonacci.py
import unittest


def fib(n):

    # Compute the nth Fibonacci number
	f=[[1,1],[1,0]]
	if n<0:
	    raise Exception("")
	if n==0:
		return 0
	if n==1:
	    return 1
	if n==2:
	    return 1
	i=1
	j=1
	for r in range(2,n):
	    k=i+j
	    i=j
	    j=k
	return k
# Tests

class Test(unittest.TestCase):

    def test_zeroth_fibonacci(self):
        actual = fib(0)
        expected = 0
        self.assertEqual(actual, expected)

    def test_first_fibonacci(self):
        actual = fib(1)
        expected = 1
        self.assertEqual(actual, expected)

    def test_second_fibonacci(self):
        actual = fib(2)
        expected = 1
        self.assertEqual(actual, expected)

    def test_third_fibonacci(self):
        actual = fib(3)
        expected = 2
        self.assertEqual(actual, expected)

    def test_fifth_fibonacci(self):
        actual = fib(5)
        expected = 5
        self.assertEqual(actual, expected)

    def test_tenth_fibonacci(self):
        actual = fib(10)
        expected = 55
        self.assertEqual(actual, expected)

    def test_negative_fibonacci(self):
        with self.assertRaises(Exception):
            fib(-1)


unittest.main(verbosity=2)